core/src/main/java/org/apache/druid/math/expr/Function.java

```diff
@@ -737,7 +745,12 @@ private ExprEval eval(ExprEval param, int scale)
       if (param.type() == ExprType.LONG) {
         return ExprEval.of(BigDecimal.valueOf(param.asLong()).setScale(scale, RoundingMode.HALF_UP).longValue());
       } else if (param.type() == ExprType.DOUBLE) {
-        return ExprEval.of(BigDecimal.valueOf(param.asDouble()).setScale(scale, RoundingMode.HALF_UP).doubleValue());
+        double val = param.asDouble();
+        if (Double.isNaN(val) || val == Double.POSITIVE_INFINITY || val == Double.NEGATIVE_INFINITY) {
+          // This is the behavior of Math.round()
+          return ExprEval.of(0L);
```

Comment:
Should we use Math.round(val) here instead of hard-code to 0?

Reply:
I wasn't sure of the perf implications of calling `Math.round` It looks like it calls out to some native function, does some bit magic and then finally just casts the object to a long - which is how we get `0L`

So I thought it would be better to just return 0 directly

Reply:
Sounds good to me.